### PR TITLE
[Merged by Bors] - refactor(Set/Countable): redefine `Set.Countable`

### PR DESCRIPTION
**Pull request by urkud**
Redefine `Set.Countable s` as `_root_.Countable s`.
Fix compile, golf some of the broken proofs.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by j-loreaux — APPROVED** [review]
I think this is nice, but I'd like a second opinion.

**Comment by j-loreaux** [comment]
maintainer merge

File: Mathlib/Data/Set/Countable.lean
Comment (eric-wieser) [inline Mathlib/Data/Set/Countable.lean:47]: Does this mean that `[S.Countable]` is now usable as a typeclass argument?

Comment (urkud) [reply Mathlib/Data/Set/Countable.lean:47]: Yes, and it means the same as `[Countable s]`.

**Review by jcommelin — APPROVED** [review]
LGTM

@eric-wieser shall we merge this?

File: Mathlib/Data/Set/Countable.lean
Comment (sgouezel) [inline Mathlib/Data/Set/Countable.lean:40]:
```suggestion
/-- A set `s` is countable if the corresponding subtype is countable,
```

**Review by sgouezel — COMMENTED** [review]
LGTM

File: Mathlib/Data/Set/Countable.lean
Comment (eric-wieser) [reply Mathlib/Data/Set/Countable.lean:47]: I'd be tempted to use the safer
```suggestion
structure Countable (s : Set α) : Prop where
  countable_coe : Countable s
```
or if that's really objectionable,
```suggestion
protected def Countable (s : Set α) : Prop := Countable s
```

Comment (urkud) [reply Mathlib/Data/Set/Countable.lean:47]: What exactly are you afraid of? I deleted `have := fun a ↦ (h a).to_subtype` or `have := h.to_subtype` here and there because it's an `abbrev` now. I can restore it but I need a reason.

Comment (urkud) [reply Mathlib/Data/Set/Countable.lean:47]: !bench

Comment (urkud) [reply Mathlib/Data/Set/Countable.lean:47]: UPD: I see that you answered on Zulip. Let's discuss there.

**Comment by urkud** [comment]
@eric-wieser I changed from `abbrev` to `def`. Could you please have another look?

File: Mathlib/Data/Set/Countable.lean
Comment (alreadydone) [inline Mathlib/Data/Set/Countable.lean:15]: no longer an abbrev ...

**Review by kmill — COMMENTED** [review]
Looks good to me.

Having it be an `abbrev` would be interesting... but that would take some design decisions for how it should appear in binders. The benefit would be that any theorem or definition requiring `[Countable s]` would automatically apply. If we went with `abbrev`, my instinct is that we should disallow `[Set.Countable s]` in binders, since I think allowing it would cause some duplication (you'd be tempted to have variations of theorems having both explicit and instance implicit binders).

bors r+

**Pull request closed without merging**
